Flaky TestStaticConfigHUPWithRotation

W0727 22:08:01.691388   25051 auth_server_static.go:202] Config parsed using legacy configuration. Please update to the latest format: {"user":[{"Password": "xxx"}, ...]}
--- FAIL: TestStaticConfigHUPWithRotation (0.02s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

--- a/go/mysql/auth_server_static_test.go
+++ b/go/mysql/auth_server_static_test.go
@@ -207,10 +207,6 @@ func hupTestWithRotation(t *testing.T, aStatic *AuthServerStatic, tmpFile *os.Fi
 		t.Fatalf("couldn't overwrite temp file: %v", err)
 	}
 
-	if aStatic.getEntries()[oldStr][0].Password != oldStr {
-		t.Fatalf("%s's Password should still be '%s'", oldStr, oldStr)
-	}
-
 	time.Sleep(20 * time.Millisecond) // wait for signal handler
 
 	if aStatic.getEntries()[oldStr] != nil {

